multisig-wallet/src/main.sw

```diff
@@ -46,28 +44,39 @@ storage {
 
     /// Number of approvals per user
     weighting: StorageMap<Address,
-    u64>, 
+    Owner>, 
 }
 
 impl MultiSignatureWallet for Contract {
     /// The constructor initializes the necessary values and unlocks further functionlity
     ///
-    /// # Panics
+    /// # Arguments
+    ///
+    /// - `users` -
+    /// - `threshold` -
+    ///
+    /// # Reverts
     ///
     /// - When the constructor is called more than once
     /// - When the user address is the 0th address (0x00000...)
     /// - When the threshold is set to 0
     /// - When an owner has an approval weight of 0
-    #[storage(read, write)]fn constructor(users: [User;
-    25], threshold: u64) {
+    #[storage(read, write)]fn constructor(users: Vec<User>, threshold: u64) {
         require(storage.nonce == 0, InitError::CannotReinitialize);
         require(storage.threshold != 0, InitError::ThresholdCannotBeZero);
 
         let mut user_index = 0;
-        while user_index < 25 {
-            require(~Address::from(BASE_ASSET_ID) != users[user_index].identity, InitError::AddressCannotBeZero);
-            require(users[user_index].weight != 0, InitError::WeightingCannotBeZero);
-            storage.weighting.insert(users[user_index].identity, users[user_index].weight);
+        while user_index < users.len() {
+            // Workaround
+            let user: Option<User> = users.get(user_index);
+            let user = user.unwrap();
+
+            require(BASE_ASSET_ID != user.identity.value, InitError::AddressCannotBeZero);
```

Comment:
Super minor nit here, but seeing BASE_ASSET_ID first in the comparison puts the emphasis in it, rather than on the `.value` field of the user. Reversing these to be
 `require(user.identity.value != BASE_ASSET_ID, InitError::AddressCannotBeZero);`
reads a bit better, but maybe it's just me.

Also, I get why you used BASE_ASSET_ID here, but creating a local const with a better name _might_ be a bit clearer, i.e:
`const ZERO_ADDRESS = BASE_ASSET_ID`

Reply:
Yeah, that should probably be added. QQ: If I add an address to `storage`, ie:
```
storage {
    user: Address,
}
```
Does `user` start as and `Address` type with it's inner value `0x000...000`?

Reply:
I know we've had this discussion around preferring to use the `Option` type over the ZERO address, where `None` would replace the usage of ZERO_ADDRESS in comparisons.

Reply:
From memory I cannot confirm since I do not recall if I have ever explicitly tested that however all `b256` values are initialized to 0x00... therefore that _should_ naturally happen to the address inner value.